custom_components/localtuya/cover.py

```diff
@@ -180,108 +201,85 @@ def __init__(
+            "Initialized tuya cover  [{}] with switch status [{}]".format(
+                self.name, self._status
             )
         )
 
     @property
-    def available(self):
-        """Return if device is available or not."""
-        return self._available
+    def is_closed(self):
+        """Check if the cover is fully closed."""
+        if self._current_cover_position == 100:
+            return True
+        return False
 
-    @property
-    def supported_features(self):
-        """Flag supported features."""
-        supported_features = (
-            SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP | SUPPORT_SET_POSITION
-        )
-        return supported_features
+    def status_updated(self):
+        """Device status was updated."""
+        _LOGGER.debug("status_updated called")
+        self._cached_status = self._status
+        _LOGGER.debug("status_updated called, self._status=%s, self._cached_status=%s", self._status, self._cached_status)
+        try:
+            self._last_movement = self._cached_status['dps'][str(self._config[CONF_LAST_MOVEMENT])]
+            self._last_position_set = self._cached_status['dps'][str(self._config[CONF_SET_POSITION])]
+            self._current_cover_position = self._cached_status['dps'][str(self._config[CONF_GET_POSITION])]
+            self._last_command = self._cached_status['dps'][str(self._dps_id)]
+        except:
+            _LOGGER.warning("status_updated failed to update certain variables")
 
     @property
     def current_cover_position(self):
-        # self.update()
-        # state = self._state
-        #        _LOGGER.info("curr_pos() : %i", self._position)
-        # print('curr_pos() : state [{}]'.format(state))
-        return self._position
+        """Return position of Tuya cover."""
+        return self._current_cover_position 
 
     @property
-    def is_opening(self):
-        # self.update()
-        state = self._state
-        # print('is_opening() : state [{}]'.format(state))
-        if state == "on":
-            return True
-        return False
+    def min_position(self):
+        """Return minimum position of Tuya cover."""
+        return MIN_POSITION
 
     @property
-    def is_closing(self):
-        # self.update()
-        state = self._state
-        # print('is_closing() : state [{}]'.format(state))
-        if state == "off":
-            return True
-        return False
+    def max_position(self):
+        """Return maximum position of Tuya cover."""
+        return MAX_POSITION 
 
     @property
-    def is_closed(self):
-        """Return if the cover is closed or not."""
-        # _LOGGER.info("running is_closed from cover")
-        # self.update()
-        state = self._state
-        # print('is_closed() : state [{}]'.format(state))
-        if state == "off":
-            return False
-        if state == "on":
-            return True
-        return None
+    def available(self):
```

Comment:
This is already in `LocalTuyaEntity`, so you can remove it here.

Reply:
👍